packages/frontend-2/components/auth/sso/Login.vue

```diff
@@ -0,0 +1,187 @@
+<template>
+  <form method="post" @submit="onSubmit">
+    <div class="flex flex-col gap-4">
+      <h1 class="text-heading-xl text-center mb-8">Speckle SSO login</h1>
+
+      <FormTextInput
+        v-model:model-value="email"
+        type="email"
+        name="email"
+        label="Your work email"
+        placeholder="Enter your email"
+        size="lg"
+        color="foundation"
+        :rules="[isEmail, isRequired]"
+        :loading="isChecking"
+        :help="helpText"
+        :custom-error-message="errorMessage"
+        show-label
+        :disabled="loading"
+        auto-focus
+        @update:model-value="onEmailChange"
+      />
+
+      <FormSelectBase
+        v-if="shouldShowWorkspaceSelector"
+        v-model="selectedWorkspace"
+        name="workspace"
+        :multiple="false"
+        label="Select workspace"
+        button-style="tinted"
+        :items="availableWorkspaces"
+        :disabled="loading"
+        placeholder="Choose a workspace"
+        help="You may need to authenticate separately for each workspace you want to access."
+        show-label
+      >
+        <template #option="{ item }">
+          <div class="flex items-center gap-2">
+            <NuxtImg v-if="item.logo" :src="item.logo" class="h-6 w-6 rounded-full" />
+            <span>{{ item.name }}</span>
+          </div>
+        </template>
+        <template #something-selected="{ value }">
+          <div v-if="isWorkspace(value)" class="flex items-center gap-2">
```

Comment:
I think the type guard is needed because the component doesn't properly type the value based on multiple: false, so TypeScript thinks it could still be an array.

Reply:
@andrewwallacespeckle That's what the `isArrayValue`/`isMultiitemArrayValue` utils off `useFormSelectChildInternals()` are for